### DISCUSSION TITLE
automate not sending emails

### DISCUSSION THREAD

**Discussion by samg666** [Budibase/budibase#6265] [Help] [answered]
Hello,

I have configured the Email SMTP on Budibase, and works fine on password recovery etc. But when I trying to send email via automate, it works whild I running the test, but didn't send email when triggered by a button click. And I didn't find any error on the Chrome console and the docker logs, didn't find any useful information. Can anyone help? 

### Below is logs on Budibase when button and triggered
`    responseTime: 56
    req: {
      "id": 25320,
      "method": "POST",
      "url": "/api/automations/au_72b6946a3c58480ba9d8e8e6300c60d5/trigger",
      "headers": {
        "connection": "upgrade",
        "x-real-ip": "192.168.100.207",
        "x-forwarded-for": "192.168.100.207",
        "host": "app-service:4002",
        "content-length": "66",
        "accept": "application/json",
        "x-budibase-app-id": "app_17fd7b3c62424e7eb26648edf1cdb771",
        "x-csrf-token": "3de30fac-e4f7-41ca-9b26-03c5ce77d4de",
        "x-budibase-api-version": "1",
        "x-budibase-type": "client",
        "user-agent": "Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/102.0.0.0 Safari/537.36",
        "content-type": "application/json",
        "origin": "http://192.168.100.14:10000",
        "referer": "http://192.168.100.14:10000/app/ops",
        "accept-encoding": "gzip, deflate",
        "accept-language": "zh-CN,zh;q=0.9",
        "cookie": "budibase:init=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpYXQiOjE2NTQ4NDkxMTd9.mUxL26zcNEF9ZWMGC0sAXp-A61ScFguMzJlOeGrymLc; budibase:init.sig=5gBn9Mu4c3gGVyoprVnAV3XXFFs; budibase:auth=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VySWQiOiJ1c19jMTliNjQyZjBlMjQ0NDdlOTMyOWU1MjA4M2ZkNzcwYiIsInNlc3Npb25JZCI6IjQ2N2EyN2M5NzczZTQ1NDZiNWNjYzk0MDhkNGVjYjE4IiwidGVuYW50SWQiOiJkZWZhdWx0IiwiaWF0IjoxNjU0ODQ5MTU1fQ.VK1IOeEl9VTo7wg4AeAQ4v1HskbxVKSwYck648zwolY; budibase:auth.sig=OFUfuINBQeGDlJnt1XHuPkJCF7k; budibase:currentapp=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJhcHBJZCI6ImFwcF8xN2ZkN2IzYzYyNDI0ZTdlYjI2NjQ4ZWRmMWNkYjc3MSIsImlhdCI6MTY1NDg1NTE5OX0.53TmK92wE5niP-EkBSUNm9MolpqxeJmNNRY0riHa4rQ"
      },
      "remoteAddress": "::ffff:172.22.0.2",
      "remotePort": 34784
    }`

### My configs
![iShot_2022-06-10_17 48 32](https://user-images.githubusercontent.com/29649651/173040802-3f57a0b0-6e3e-4aed-acf2-f4b10fb028c5.png)

![iShot_2022-06-10_17 50 05](https://user-images.githubusercontent.com/29649651/173040822-63891483-0b6f-4d61-9a14-d20b82e87440.png)

**Comment by melohagan** [comment]
Hi @ggs331 

Are you clicking the button in your published app?

I.e. the URL in your browser is in the form: `/app/appName` and NOT `/app_dev_etc`

**Reply by samg666** [reply]
@melohagan Yes, on the published apps. I know that automate is not working on the dev mode, everytime are on published app. But still not working, and I don't know what else I can debug now?

**Comment by aptkingston** [comment]
Hey @ggs331. I tested this myself, using exactly the same structure for everything that you have - but for me everything works correctly so I'm quite confused why it doesn't work for you.

Here is the setup that works for me:
![image](https://user-images.githubusercontent.com/9075550/173087565-65c8de34-9464-417b-8069-e1873c7a9041.png)
![image](https://user-images.githubusercontent.com/9075550/173087645-d8290081-6f7c-4c1b-bd9b-7faf2f795087.png)
![image](https://user-images.githubusercontent.com/9075550/173086630-9c498013-2012-4779-8f41-d9b6b55823d6.png)
![image](https://user-images.githubusercontent.com/9075550/173087832-952e98dd-c59e-4f2b-a976-2cded479d5bb.png)

Do you get the success popup when you trigger the automation?

Could you also check your browser network tab and see if the response from the server looks correct? It should look something like this:
![image](https://user-images.githubusercontent.com/9075550/173087186-5deba93b-b937-449e-b195-dabe7fa00041.png)

The only thing I can suggest is try deleting that button action and create it again. If you created your automation a few times and deleted it, then the automation ID referenced in the button action might be stale - so if you delete and recreate the button action then it will make sure it's the correct automation ID. This might not change anything, but it's worth trying.

**Reply by samg666** [reply]
Hi @aptkingston , I have just re-created and tested still the same. The email can be sent on the test mode, but still not work on published app. I have do some debug on network layer, there's no any network traffic sending to stmp.gmail.com port 465(I am using Gmail) when triggered on publised. But I can see the network traffic when send on test mode. So, it should not be the networking issue between smtp server, it seems the Budibase didn't actually triggered the action. Is there any other way to do the debugging?

### The automation triggered success, and I can see the logs on docker/budibase/apps
<img width="2221" alt="image" src="https://user-images.githubusercontent.com/29649651/173094956-a44c657f-64e4-42ab-b3f3-2fbde72589bb.png">

### I can receive email when on the test mode
<img width="1616" alt="iShot_2022-06-10_23 18 22" src="https://user-images.githubusercontent.com/29649651/173097420-4ad84518-1581-4b32-a96f-a4480a31b258.png">

### Debug on email sending network activities, no any network traffic when triggerred on published app, but I can see the traffic when sending on test mode.
<img width="1446" alt="image" src="https://user-images.githubusercontent.com/29649651/173098308-aac82253-7cc7-4337-b4f3-cec855d30a38.png">

**Comment by samg666** [comment, marked as answer]
The email issue has been resolved after upgraded from v1.0.184 to v1.0.206. There might has some bugs on v1.0.184. Anyway, it's fixed after upgrade. Thanks all.
